lib/logstash/outputs/scalyr.rb

```diff
@@ -333,19 +353,58 @@ def build_multi_event_request_array(logstash_events)
         end
       end
 
-      # Rename user-specified origin field -> 'origin'
-      rename.call(@origin_field, 'origin')
+      # Rename user-specified serverHost field -> 'serverHost'
+      rename.call(@serverhost_field, 'serverHost')
+      record.delete(@serverhost_field)
 
       # Rename user-specified logfile field -> 'logfile'
       rename.call(@logfile_field, 'logfile')
+      record.delete(@logfile_field)
       # Set logfile field if empty and origin is supplied
       if record['logfile'].to_s.empty? and origin
         record['logfile'] = "/logstash/#{origin}"
       end
 
+      log_identifier = nil
+      add_log = false
+      if origin
+       log_identifier = origin + record['logfile']
+      end
+      if log_identifier and not logs.key? log_identifier
+        add_log = true
+        logs[log_identifier] = {
+          'id' => next_log_id,
+          'attrs' => Hash.new
+        }
+        if not record['serverHost'].to_s.empty?
+          logs[log_identifier]['attrs']['serverHost'] = record['serverHost']
+          record.delete('serverHost')
+        end
+        if not record['logfile'].to_s.empty?
+          logs[log_identifier]['attrs']['logfile'] = record['logfile']
+          record.delete('logfile')
+        end
+        if @log_constants
+          @log_constants.each {|log_constant|
+            if record.key? log_constant
+              logs[log_identifier]['attrs'][log_constant] = record[log_constant]
+            end
+          }
```

Comment:
As mentioned in another comment parsers are expected to be defined by the user in Logstash filters, but they might not necessarily exist at all, nor be constant for a logfile if the user doesn't want it, so it isn't defaulted to being a log level attribute.

Reply:
Under what cases would the parser not be constant for a log file?

Reply:
With Logstash filters the user can add or modify fields on a per-message basis, so it is entirely possible they would set different parsers based on different message contents instead of logfile, or any other field really.